pkg/verification/verify.go

```diff
@@ -18,17 +18,220 @@ package verification
 import (
 	"bytes"
 	"crypto/x509"
-	"errors"
+	"encoding/asn1"
+	"encoding/pem"
 	"fmt"
 	"hash"
 	"io"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
 
 	"github.com/digitorus/pkcs7"
 	"github.com/digitorus/timestamp"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
 )
 
+type VerifyOpts struct {
+	Oid            asn1.ObjectIdentifier
+	TsaCertificate *x509.Certificate
+	Intermediates  []*x509.Certificate
+	Roots          []*x509.Certificate
+	Nonce          *big.Int
+	Subject        string
+	HashAlgorithm  hash.Hash
+	HashedMessage  []byte
+}
+
+func NewVerifyOpts() (VerifyOpts, error) {
+	opts := VerifyOpts{}
+
+	oidFlagVal := viper.GetString("oid")
```

Comment:
Sorry, I might have misunderstood, but we should keep flags out of the core library and just in the client. If you want to have this function that handles things like parsing OIDs, it should take in values that are already read in from the flags. Also, personally I prefer to not have files read from disk as part of the library, so I think this should also be handled in the client and the chain passed into the function.

So I'd say either:
* Move NewVerifyOpts to the client application with all flag parsing there
* Move NewVerifyOpts to the client app, but provide some helper utilities for constructing OIDs and splitting PEM chains into cert pools
* Keep NewVerifyOpts here, but have it take in all values (OID as a string, PEM chain as a string), and read in flags in the client app

Does that seem reasonable?

Reply:
I agree about keeping the flag parsing in the `app` CLI package. Just moved that over.